### PR TITLE
Psalm 4

### PR DESCRIPTION
**Pull request by franmomu**
## Subject

Raises Psalm to level 4 fixing some types on the way

I am targeting this branch, because fixes some phpdoc types.

## Changelog

```markdown
### Fixed
- Fixed some phpdoc types
```

### CODE REVIEW

File: src/EventListener/LogRevisionsListener.php [hunk 1]
```diff
@@ -308,7 +308,7 @@ private function getOriginalEntityData(object $entity): array
     }
 
     /**
-     * @return string|int
+     * @return string|int|false
```
Comment (VincentLanglet) [inline src/EventListener/LogRevisionsListener.php]: if you're returning false, it means you're setting false to the property too.

Also `false` doesn't seems to be a supported return type because it's later use as a param in the query.

Maybe we should throw an exception if the value is false ?

File: src/AuditReader.php [hunk 2]
```diff
@@ -571,7 +571,7 @@ public function findRevisions($className, $id)
      * @throws NotAuditedException
      * @throws Exception
      *
-     * @return int|string
+     * @return int|string|false
```
Comment (VincentLanglet) [inline src/AuditReader.php]: should we prefer null ?
or throw an exception in next major ?

File: src/Collection/AuditedCollection.php [hunk 3]
```diff
@@ -122,7 +122,7 @@ public function __construct(AuditReader $auditReader, $class, ClassMetadataInfo
     }
 
     /**
-     * @return bool
+     * @return true
```
Comment (VincentLanglet) [inline src/Collection/AuditedCollection.php:125]: why this change ? this method return nothing.

File: src/Metadata/MetadataFactory.php [hunk 4]
```diff
@@ -18,12 +18,12 @@ class MetadataFactory
     /**
      * @var string[]
      *
-     * @phpstan-var class-string[]
+     * @phpstan-var array<class-string, int|string>
```
Comment (VincentLanglet) [inline src/Metadata/MetadataFactory.php]: array<class-string, int|string> is not fully compatible with string[], so maybe the @var need to be updated too.

File: src/Metadata/MetadataFactory.php [hunk 5]
```diff
@@ -18,12 +18,12 @@ class MetadataFactory
...
      */
     private $auditedEntities = [];
 
     /**
-     * @phpstan-param array<class-string, mixed> $auditedEntities
+     * @phpstan-param class-string[] $auditedEntities
```
Comment (VincentLanglet) [inline src/Metadata/MetadataFactory.php:26]: there is something weird according to phpstan

File: src/Metadata/MetadataFactory.php [hunk 6]
```diff
@@ -16,36 +16,36 @@
...
      */
     public function __construct(array $auditedEntities)
     {
-        $this->auditedEntities = array_flip(array_filter($auditedEntities, static function ($record): bool {
-            return \is_string($record) || \is_int($record);
-        }));
+        $this->auditedEntities = array_flip($auditedEntities);
```
Comment (franmomu) [inline src/Metadata/MetadataFactory.php]: Apparently this was done in https://github.com/sonata-project/EntityAuditBundle/issues/143 because of SonataDoctrineORMAdminBundle, but [looking at the tests there](https://github.com/sonata-project/SonataDoctrineORMAdminBundle/pull/1652), looks like there are no `null` values. ~We could still add an `array_filter` call just in case~, but if there are `null` values, we should fix it in the doctrine orm admin bundle.

File: src/Collection/AuditedCollection.php (hunk 3 above)
Comment (franmomu) [reply src/Collection/AuditedCollection.php:125]: To have the same return type as the interface: https://github.com/doctrine/collections/blob/c1ec9823f9250fefb274cccf62970b5c167e6bc7/lib/Doctrine/Common/Collections/Collection.php#L34-L42

File: src/AuditReader.php (hunk 2 above)
Comment (franmomu) [reply src/AuditReader.php]: I'll do that

**Review by VincentLanglet — APPROVED** [review]
